drivers/bluetooth/hci/hci_nxp.c

```diff
@@ -47,7 +54,29 @@ LOG_MODULE_REGISTER(bt_driver);
 #define HCI_CMD_SET_BT_SLEEP_MODE_PARAM_LENGTH          3U
 #define HCI_CMD_BT_HOST_SLEEP_CONFIG_OCF                0x59U
 #define HCI_CMD_BT_HOST_SLEEP_CONFIG_PARAM_LENGTH       2U
+#define HCI_CMD_BT_HOST_SET_MAC_ADDR_PARAM_LENGTH       8U
+#define HCI_SET_MAC_ADDR_CMD                            0x0022U
+#define BT_USER_BD                                      254
+#define BD_ADDR_OUI                                     0x37U, 0x60U, 0x00U
+#define BD_ADDR_OUI_PART_SIZE                           3U
+#define BD_ADDR_UUID_PART_SIZE                          3U
+
+#if !defined(CONFIG_HCI_NXP_SET_CAL_DATA)
+#define bt_nxp_set_calibration_data() 0
+#endif
+
+#if !defined(CONFIG_HCI_NXP_SET_CAL_DATA_ANNEX100)
+#define bt_nxp_set_calibration_data_annex100() 0
+#endif
+
+#if !defined(CONFIG_HCI_NXP_ENABLE_AUTO_SLEEP)
+#define nxp_bt_set_host_sleep_config()       0
+#define nxp_bt_enable_controller_autosleep() 0
+#endif
 
+#if !defined(CONFIG_BT_HCI_SET_PUBLIC_ADDR)
+#define bt_nxp_set_mac_address(public_addr) 0
+#endif
```

Comment:
the custom is to put all these empty defines in an #else block right below where they would otherwise be defined, see other networking files for example

non-blocking suggestion